tests/test_import_export.py

```diff
@@ -0,0 +1,203 @@
+        """
+        bundle = XBundle(keep_urls=True, keep_studio_urls=True)
+        bundle.import_from_directory(os.path.join('input_testdata', 'mitx.01'))
+
+        bundle_string = str(bundle)
+
+        expected = expected_data.KEEP_URLS
+        self.assertEqual(clean_xml(expected), clean_xml(bundle_string))
+
+    def test_preserve_url_name(self):
+        """
+        Test that preserve_url_name imports as url_name and not url_name_orig.
+        """
+        bundle = XBundle(
+            keep_urls=True, keep_studio_urls=True, preserve_url_name=True)
+        bundle.import_from_directory('input_testdata/mitx.01')
+
+        bundle_string = str(bundle)
+
+        expected = expected_data.PRESERVE_URL_NAME
+        self.assertEqual(clean_xml(expected), clean_xml(bundle_string))
+
+    def test_save(self):
+        """
+        Test save method.
+        """
+
+        input_xml = "<xbundle><metadata /><course /></xbundle>"
+        bundle = XBundle()
+        bundle.load(file_from_string(input_xml))
+        self.assertEqual(clean_xml(str(bundle)), clean_xml(input_xml))
+
+        curdir = os.getcwd()
+        tempdir = mkdtemp()
+        try:
+            os.chdir(tempdir)
+            bundle.save()
+
+            with open(os.path.join(tempdir, "xbundle.xml")) as f:
+                self.assertEqual(clean_xml(f.read()), clean_xml(input_xml))
+
+            bundle.save(filename="other.xml")
+            with open(os.path.join(tempdir, "other.xml")) as f:
+                self.assertEqual(clean_xml(f.read()), clean_xml(input_xml))
+
+            handle_path = os.path.join(tempdir, "third.xml")
+            with open(handle_path, "w") as f:
+                bundle.save(file_handle=f)
+            with open(handle_path) as f:
+                self.assertEqual(clean_xml(f.read()), clean_xml(input_xml))
+        finally:
+            os.chdir(curdir)
+            rmtree(tempdir)
+
+    def test_export_and_keep_urls(self):
+        """
+        Test the changes to url_name after export_to_directory and import.
+        """
+        # Note url_name_orig in chapter.
+        input_xml = input_data.URL_NAME_ORIG_IN_CHAPTER2
+        bundle = XBundle(keep_urls=True, force_studio_format=True)
+        bundle.load(file_from_string(input_xml))
+
+        # str(bundle) doesn't change input xml, but export_to_directory will.
+        self.assertEqual(clean_xml(input_xml), clean_xml(str(bundle)))
+
+        old_current_dir = os.getcwd()
+        tempdir = mkdtemp()
+        try:
+            os.chdir(tempdir)
+            bundle.export_to_directory()
+
+            bundle2 = XBundle(keep_urls=True, force_studio_format=True)
+            bundle2.import_from_directory()
+
+            expected = expected_data.KEEP_URLS_FORCE_STUDIO_FORMAT
+
+            self.assertEqual(clean_xml(expected), clean_xml(str(bundle2)))
+        finally:
+            os.chdir(old_current_dir)
+            rmtree(tempdir)
+
+    def test_xml_header(self):
```

Comment:
I'm missing what this is testing; it seems to be comparing `input_xml` to itself. Maybe it just needs a more verbose comment.